### PR TITLE
Added any and all helpers

### LINKED ISSUE #2021: Expand helper js to include ifor and ifand
Update the core helper js to include ifor and ifand statements.

Example of ifor use:
Only render the header section of an article / block / component when **either** the displayTitle, body, or instruction is present. 

```
{{#any displayTitle body instruction}}
<div class="component__header {{_component}}__header">
</div>
{{/any}}
```

Example of ifand use:
Only render the header section of an article / block / component when the displayTitle, body, and instruction are **all** present.

```
{{#all displayTitle body instruction}}
<div class="component__header {{_component}}__header">
</div>
{{/all}}
```

Example of ifand ifor use:
```
{{#all (any displayTitle title) body instruction}}
<div class="component__header {{_component}}__header">
</div>
{{/all}}
```

### PR DESCRIPTION
**Pull request by guywillis**
resolves #2021

### CODE REVIEW

**Comment by tomgreenfield** [comment]
Suggest some more readable helper names? I keep reading as i-for and i-fand.

Possibly:
```{{#either ...}}```
```{{#all ...}}```

File: src/core/js/helpers.js [hunk 1]
```diff
@@ -67,6 +67,28 @@ define([
             }
         },
 
+        ifor: function() {
+            var args = Array.prototype.slice.call(arguments, 0, -1);
+            var block = Array.prototype.slice.call(arguments, -1)[0];
+
+            for (var i = 0, l = args.length; i < l; i++) {
+                if (args[i]) return block.fn(this);
+            }
```
Comment (oliverfoster) [inline src/core/js/helpers.js]: use:

```js
return _.some(args) ? block.fn(this) : block.inverse(this);
```

File: src/core/js/helpers.js [hunk 2]
```diff
@@ -67,6 +67,28 @@ define([
...
+            return block.inverse(this);
+        },
+
+        ifand: function() {
+            var args = Array.prototype.slice.call(arguments, 0, -1);
+            var block = Array.prototype.slice.call(arguments, -1)[0];
+
+            for (var i = 0, l = args.length; i < l; i++) {
+                if (!args[i]) return block.inverse(this);
+            }
```
Comment (oliverfoster) [inline src/core/js/helpers.js]:
```js
return args.every() ? block.fn(this) : block.inverse(this);
```

File: src/core/js/helpers.js [hunk 3]
```diff
@@ -67,6 +67,28 @@ define([
             }
         },
 
+        ifor: function() {
```
Comment (oliverfoster) [inline src/core/js/helpers.js]: rename to `any`
add comment

File: src/core/js/helpers.js [hunk 4]
```diff
@@ -67,6 +67,28 @@ define([
...
+            var block = Array.prototype.slice.call(arguments, -1)[0];
+
+            for (var i = 0, l = args.length; i < l; i++) {
+                if (args[i]) return block.fn(this);
+            }
+
+            return block.inverse(this);
+        },
+
+        ifand: function() {
```
Comment (oliverfoster) [inline src/core/js/helpers.js]: rename to `all`
add comment

Comment (oliverfoster) [reply src/core/js/helpers.js]:
```js
return args.every() ? block.fn(this) : block.inverse(this);
```

File: src/core/js/helpers.js [hunk 5]
```diff
@@ -67,6 +67,26 @@ define([
             }
         },
 
+        /**
+         * if (conditionA || conditionB)
+         */
```
Comment (moloko) [inline src/core/js/helpers.js]: could you amend to:
```
/**
 * Equivalent to 
 * @example
 * {{#any displayTitle body instruction}}
 * <div class="component__header {{_component}}__header"></div>
 * {{/any}}
 */
```

Comment (moloko) [reply src/core/js/helpers.js]: and something similar for `any`?

**Comment by guywillis** [comment]
Amended comments

**Comment by chetan-hajare** [comment]
Since the content of both the functions are same except(_.any and _.all), can we have a common function and pass "any"/"all" as parameter? 
Otherwise I am fine with current structure as well.

**Pull request merged**